src/lvl_script_commands.c

```diff
@@ -2030,7 +2030,10 @@ static void set_room_configuration_process(struct ScriptContext *context)
             roomst->update_workers_in_room = terrain_room_used_capacity_func_list[value2];
             reinitialise_rooms_of_kind(room_type);
             break;
-        case 16: // StorageHeight
+        case 16: // SlabSynergy
+            roomst->synergy_slab = value;
+            break;
```

Comment:
I think NULL would work out of the box because it is room0 that is called like that.

Reply:
synergy_slab is a slab, not a room. There is no NULL slab? I've been using -3 and -2 for NULL/none.

 The full code for the SlabAssign that I could see is:
        case 10: // SlabAssign
            roomst->assigned_slab = value;
            break;
            
I don't see the else if?

Reply:
set_room_configuration_check seems to be dealing in numeric ids, though?
Again, for case 10 there is no check that I can see, it simply assigns the number to the id directly. Could you clarify where these checks would be appropriate, maybe referring me to where they already exist for the SlabAssign field so that I can base my implementation on the existing one.

Reply:
I just made a commit with the code you sent. The code that handles the script sounds a bit complex, I think it'd be a good idea to test roomvar 16 and 17. To be honest, I haven't yet tried making a map script, and I don't know exactly what StorageHeight does.